src/decoding.cc

```diff
@@ -197,9 +199,31 @@ namespace ctranslate2 {
       topk_log_probs.reshape({cur_batch_size, _beam_size});
       topk_scores.reshape({cur_batch_size, _beam_size});
       topk_ids.reshape({cur_batch_size, _beam_size});
+
+      if(_coverage_penalty != 0){
```

Comment:
The coverage penalty implementation does not support GPU execution. That's fine, but can you throw an exception if coverage penalty is enabled and the device is not the CPU?

Reply:
Now, the implementation supports GPU, I have made test on the GPU machine.